Shared/Samples/Navigate route/NavigateRouteView.swift

```diff
@@ -0,0 +1,354 @@
+            guard let routeTracker else { return }
+            
+            routeTracker.voiceGuidanceUnitSystem = Locale.current.usesMetricSystem ? .metric : .imperial
+            
+            // Gets the route and its geometry from the route result.
+            guard let firstRoute = routeResult.routes.first,
+                  let routeGeometry = firstRoute.geometry else {
+                return
+            }
+            
+            // Updates the directions.
+            directions = firstRoute.directionManeuvers
+            
+            // Creates the mock data source from the route's geometry.
+            let densifiedRoute = GeometryEngine.geodeticDensify(
+                routeGeometry,
+                maxSegmentLength: 50,
+                lengthUnit: .meters,
+                curveType: .geodesic
+            ) as! Polyline
+            let mockDataSource = SimulatedLocationDataSource(polyline: densifiedRoute)
+            
+            // Creates a route tracker location data source.
+            let routeTrackerLocationDataSource = RouteTrackerLocationDataSource(
+                routeTracker: routeTracker,
+                locationDataSource: mockDataSource
+            )
+            
+            // Sets the location display's data source.
+            locationDisplay.dataSource = routeTrackerLocationDataSource
+            
+            // Updates the graphics and viewpoint.
+            routeRemainingGraphic.geometry = routeGeometry
+            viewpoint = Viewpoint(boundingGeometry: routeGeometry)
+        }
+        
+        /// Start monitoring multiple asynchronous streams of information.
+        private func startTracking() async {
+            await withTaskGroup(of: Void.self) { group in
+                group.addTask { await self.trackAutoPanMode() }
+                group.addTask { await self.trackStatus() }
+                group.addTask { await self.trackVoiceGuidance() }
+            }
+        }
+        
+        /// Monitor the asynchronous stream of tracking statuses.
+        ///
+        /// When new statuses are delivered, update the route's traversed and remaining graphics.
+        private func trackStatus() async {
+            guard let routeTracker else { return }
+            for try await status in routeTracker.$trackingStatus {
+                if let status {
+                    routeTraversedGraphic.geometry = status.routeProgress.traversedGeometry
+                    routeRemainingGraphic.geometry = status.routeProgress.remainingGeometry
+                    
+                    switch status.destinationStatus {
+                    case .notReached, .approaching:
+                        statusText = """
+                        Distance remaining: \(status.routeProgress.remainingDistance.distanceRemainingText)
+                        Time remaining: \(timeFormatter.string(from: status.routeProgress.remainingTime) ?? "")
```

Comment:
```suggestion
                        Time remaining: \(timeFormatter.string(from: status.routeProgress.remainingTime)!)
```
From the doc I think it is safe to force unwrap.